src/lib.rs

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2024 Finnegan Dion-Kuhn
+
+use chrono::{Datelike, Local, TimeZone};
+use model::PulseData;
+use rand::{seq::SliceRandom, thread_rng};
+use tba_openapi_rust::{
+    apis::{configuration::Configuration, event_api},
+    models::event_ranking_rankings_inner::EventRankingRankingsInner,
+};
+
+mod model;
+
+pub use model::{NexusEventStatus, NexusMatch, SlideData, StatboticsReturn, StatboticsTeamYear};
+
+pub fn get_slide<'a>(
+    slides: &'a Vec<impl Fn(&SlideData) -> Option<&'a str>>,
+    slide_data: SlideData,
+) -> String {
+    match slides.choose(&mut thread_rng()).unwrap()(&slide_data) {
+        Some(slide) => String::from(slide),
+        None => get_slide(slides, slide_data),
+    }
+}
+
+pub async fn get_statbotics_data(
+    client: reqwest::Client,
+    team_key: &u16,
+    event_key: &str,
+) -> StatboticsReturn {
+    let data = client
+        .get(format!(
+            "https://api.statbotics.io/v3/team_year/{}/{}",
+            team_key,
+            chrono::Utc::now().year()
+        ))
+        .header("accept", "application/json")
+        .send()
+        .await
+        .unwrap()
+        .json::<StatboticsTeamYear>()
+        .await
+        .unwrap();
```

Comment:
You probably should match on parsing external API results to a specific struct, in case there's an issue with the query, or the spec changes or something